src/main/groovy/org/transmartproject/core/dataquery/highdim/HighDimensionDataTypeResource.groovy

```diff
@@ -28,6 +29,8 @@ interface HighDimensionDataTypeResource<R extends DataRow<AssayColumn, ? /* depe
      */
     String getDataTypeDescription()
 
+    List<OntologyTerm> getAllOntologyTermsForDataTypeBy(long resultInstanceId)
```

Comment:
* Avoid using opaque numeric identifiers when you can use type safe object references. In this case, passing a `QueryResult`.
* Result should be a set, since order doesn't matter and the same ontology term cannot be repeated.